src/Microsoft.AspNet.OData.Shared/Builder/EdmModelHelperMethods.cs

```diff
@@ -321,7 +328,12 @@ private static void ValidateOperationEntitySetPath(IEdmModel model, IEdmOperatio
                     model.SetAnnotationValue(operation, new ReturnedEntitySetAnnotation(operationConfiguration.NavigationSource.Name));
                 }
 
-                AddOperationParameters(operation, operationConfiguration, edmTypeMap);
+                if (operationConfiguration.DerivedTypeConstraint != null && operationConfiguration.DerivedTypeConstraint.Any())
```

Comment:
>&& operationConfiguration.DerivedTypeConstraint.Any() [](start = 73, length = 53)

this is already handled in AddDerivedTypeConstraintAnnotation; don't need to check twice.